### PR TITLE
Update frontend README

### PR DESCRIPTION
**Pull request by blackzlq**
1. Remove Cloud run related information.
2. Add detailed description for DLP and NLP

### CODE REVIEW

File: applications/rag/frontend/README.md [hunk 1]
```diff
@@ -3,98 +3,21 @@
 This directory contains the code for a frontend flask webserver integrating with the inference
 backend and (NYI) vector database through [LangChain](https://python.langchain.com/docs/get_started/introduction).
```
Comment (bjornsen) [inline applications/rag/frontend/README.md:4]: What is NYI?

File: applications/rag/frontend/README.md [hunk 2]
```diff
@@ -3,98 +3,21 @@
...
 
-## Cloud Run Setup
+The image is hosted on `us-central1-docker.pkg.dev/ai-on-gke/rag-on-gke` and used by the frontend deployment.
```
Comment (bjornsen) [inline applications/rag/frontend/README.md:6]: Let's take out specifying which image is used. This'll be documented in the code.

File: applications/rag/frontend/README.md [hunk 3]
```diff
@@ -3,98 +3,21 @@
...
 
-1. Configure inference internal loadbalancer service:
-```
-export INFERENCE_DEPLOYMENT=mistral-7b-instruct  # Customize as needed
-kubectl create service loadbalancer $INFERENCE_DEPLOYMENT --tcp=80:8080 && \
-kubectl annotate service/$INFERENCE_DEPLOYMENT "cloud.google.com/load-balancer-type=Internal"
+## Sensitive Data Protection
+Data Loss Prevention (DLP) and Natural Language Protection (NLP) System. Our integrated solution leverages the latest in data protection and language understanding technologies to safeguard sensitive information and ensure content appropriateness across digital platforms.
```
Comment (bjornsen) [inline applications/rag/frontend/README.md:9]: Let's refer to Cloud Natural Language API instead and link through to it. NLP as listed here isn't a product and isn't a part of SDP. Link to DLP and SDP pages so people can learn more.

File: applications/rag/frontend/README.md [hunk 4]
```diff
@@ -3,98 +3,21 @@
...
+### Features:
 
-2. Configure environment variables
-```
-# Change these values to match your desired Cloud Run deployment
-export PROJECT_ID=my-project
-export PROJECT_NUM=$(gcloud projects describe $PROJECT_ID --format="value(projectNumber)")
-export REGION=us-central1
+#### Data Loss Prevention (DLP): 
+Identifies, monitors, and protects sensitive data through detection techniques, ensuring compliance with data protection regulations.
```
Comment (bjornsen) [inline applications/rag/frontend/README.md:14]: Rewrite this specific to how we've implemented DLP in the demo. For example, note that it redacts potentially sensitive information from outputs (and data if we get to implementing that). We don't currently support any compliance regimes with our implementation here.

File: applications/rag/frontend/README.md [hunk 5]
```diff
@@ -3,98 +3,21 @@
...
-export PROJECT_NUM=$(gcloud projects describe $PROJECT_ID --format="value(projectNumber)")
-export REGION=us-central1
+#### Data Loss Prevention (DLP): 
+Identifies, monitors, and protects sensitive data through detection techniques, ensuring compliance with data protection regulations. 
 
-export INSTANCE_CONNECTION_NAME=my-project:us-central1:pgvector-instance
-export DB_USER=<user name>
-export DB_PASSWORD=<password>
-```
+#### Natural Language Proctection (NLP):
```
Comment (bjornsen) [inline applications/rag/frontend/README.md:16]: Nit: Proctection -> Protection.

Bigger discussion: NLP usually means natural language processing. I haven't run across NLP as Natural Language Protection before. Is that a new term or is it listed elsewhere? If a new term, I'd stick to using our specific product terminology around text moderation and Cloud Natural Language API.

File: applications/rag/frontend/README.md [hunk 6]
```diff
@@ -3,98 +3,21 @@
...
-export REGION=us-central1
+#### Data Loss Prevention (DLP): 
+Identifies, monitors, and protects sensitive data through detection techniques, ensuring compliance with data protection regulations. 
 
-export INSTANCE_CONNECTION_NAME=my-project:us-central1:pgvector-instance
-export DB_USER=<user name>
-export DB_PASSWORD=<password>
-```
+#### Natural Language Proctection (NLP): 
+Using  the text moderation API from Cloud Natural Language API to analyzes text for sentiment, context, and content categorization, filtering out inappropriate material based on predefined categories such as Health, Finance, Politics, and Legal.
```
Comment (bjornsen) [inline applications/rag/frontend/README.md:17]: Good content here! Quick rewrite:

"Use text moderation capabilities from the Cloud Natural Language API to analyze user input and LLM responses for sentiment, content, and content categorization. Filter out inappropriate material based on predefined categories such as health, finance, politics, and legal."

File: applications/rag/frontend/README.md [hunk 7]
```diff
@@ -104,4 +27,10 @@ We have two ways to enable the api:
 This filter can auto fetch the templates in your project. Please refer to the following links to create templates:
 
     1. Inspect templates: https://cloud.google.com/dlp/docs/creating-templates-inspect
-    2. De-identification templates: https://cloud.google.com/dlp/docs/creating-templates-deid
\ No newline at end of file
+    2. De-identification templates: https://cloud.google.com/dlp/docs/creating-templates-deid
+
+2. Enable Cloud Nature Language Protection NLP
```
Comment (bjornsen) [inline applications/rag/frontend/README.md:32]: Nit: Cloud Nature Language Protection NLP ->  Cloud Natural Language API

File: applications/rag/frontend/README.md [hunk 8]
```diff
@@ -104,4 +27,10 @@ We have two ways to enable the api:
...
+
+Follow the instruction to [enable Nature Language API](https://cloud.google.com/natural-language/docs/setup#api)
+
+The NLP (Nature Language Protection) System uses advanced NLP technologies to filter and moderate text, excluding content related to Health, Finance, Politics, and Legal categories to ensure appropriateness.
```
Comment (bjornsen) [inline applications/rag/frontend/README.md:36]: Drop this as we covered it above.

File: applications/rag/frontend/README.md [hunk 9]
```diff
@@ -104,4 +27,10 @@ We have two ways to enable the api:
 This filter can auto fetch the templates in your project. Please refer to the following links to create templates:
 
     1. Inspect templates: https://cloud.google.com/dlp/docs/creating-templates-inspect
-    2. De-identification templates: https://cloud.google.com/dlp/docs/creating-templates-deid
\ No newline at end of file
+    2. De-identification templates: https://cloud.google.com/dlp/docs/creating-templates-deid
+
+2. Enable Cloud Nature Language Protection NLP
+
+Follow the instruction to [enable Nature Language API](https://cloud.google.com/natural-language/docs/setup#api)
```
Comment (bjornsen) [inline applications/rag/frontend/README.md:34]: Nit: Follow these instructions to enable the Cloud Natural Language API

**Review by bjornsen — APPROVED** [review]

File: applications/rag/frontend/README.md (hunk 1 above)
Comment (blackzlq) [reply applications/rag/frontend/README.md]: Not sure. Removed

File: applications/rag/frontend/README.md (hunk 6 above)
Comment (blackzlq) [reply applications/rag/frontend/README.md]: Done!

**Pull request merged**